botto/tld_botto.py

```diff
@@ -778,16 +779,43 @@ async def calculate_meal_reminders(
         reminder_text = "\n".join(reminder_list)
         return f"{intro_text}\n{reminder_text}"
 
-    async def send_meal_reminder(self, reply_to: Optional[Message] = None):
+    async def remove_old_meal_reminders(self, channel: MessageableChannel):
+        meal_reminder_hours = self.config.get("meals", {}).get("auto_reminder_hours")
+        if not meal_reminder_hours:
+            return
+        previous_messages = await channel.history(limit=1).flatten()
+        if len(previous_messages) < 1:
+            return
+        last_message = previous_messages[0]
+        last_message_from_self = previous_messages[0].author.id == self.user.id
+        # This is a bit of a bodge, but we're basically trying to determine "Was this an automated reminder?"
```

Comment:
Would it be easy to store the message id of the last sent meal reminder somewhere and then just compare it?

Reply:
It would, but we don’t really have a persistent place in memory for this sort of thing